### PR TITLE
feat: Adds annotation support

### LINKED ISSUE #3: Add annotations support
Add annotations support similar to https://marshmallow-annotations.readthedocs.io/en/latest/quickstart.html

But discuss in the team first if we actually want this.

### PR DESCRIPTION
**Pull request by Panaetius**
Adds support for annotations.

Used like:

```
       import datetime.datetime as dt

       from calamus import JsonLDAnnotation
       import calamus.fields as fields

       schema = fields.Namespace("http://schema.org/")

       class User(metaclass=JsonLDAnnotation):
           class Meta:
               rdf_type = schema.Person
           _id = fields.Id()
           birth_date = fields.Date(schema.birthDate, default=dt.now)
           name = fields.String(schema.name, default=lambda: "John")

        user = User()

        # dumping
        User.schema().dump(user)
        # or
        user.dump()

        # loading
        u = User.schema().load({"_id": "http://example.com/user/1", "name": "Bill", "birth_date": "1970-01-01 00:00"})
```

Closes #3

### CODE REVIEW

**Comment by rokroskar** [comment]
I don't understand the usage here - in your example does `User` become an annotation that you can add to other objects?

**Comment by Panaetius** [comment]
No user is the object, but by declaring the attributes as calamus fields, the metaclass uses them to construct a Schema and returns the class with plain fields defined.

```
class User(metaclass=JsonLDAnnotation):
           class Meta:
               rdf_type = schema.Person
           _id = fields.Id(default=None)
           birth_date = fields.Date(schema.birthDate, default=dt.now)
```
gives you a class that looks like
```
class User:
    _id = None
    birth_date = dt.now
```
whose `.schema()` function returns a schema
```
class UserSchema:
    _id = fields.Id()
    birth_date = fields.Date(schema.birthDate)
    class Meta:
        model = User
        rdf_type = schema.Person
```

**Comment by rokroskar** [comment]
Oops - I read "annotation" and immediately though of https://www.w3.org/ns/oa - very different annotation.

**Review by rokroskar — CHANGES_REQUESTED** [review]
This looks great! could you add an example to the readme?

File: README.rst [hunk 1]
```diff
@@ -119,6 +120,38 @@ You can also easily deserialize JSON-LD to python objects
...
+    import calamus.fields as fields
+
+    schema = fields.Namespace("http://schema.org/")
+
+    class User(metaclass=JsonLDAnnotation):
+        class Meta:
+            rdf_type = schema.Person
+        _id = fields.Id()
+        birth_date = fields.Date(schema.birthDate, default=dt.now)
+        name = fields.String(schema.name, default=lambda: "John")
```
Comment (rokroskar) [inline README.rst:143]: For ease of comparison, I would keep the ordering here the same as for `BookSchema` above:

```suggestion
        _id = fields.Id()
        birth_date = fields.Date(schema.birthDate, default=dt.now)
        name = fields.String(schema.name, default=lambda: "John")
        
        class Meta:
            rdf_type = schema.Person
```

**Review by rokroskar — APPROVED** [review]
Minor suggestion - feel free to ignore. 👍

**Review by rokroskar — DISMISSED** [review]

**Review by rokroskar — APPROVED** [review]

**Pull request merged**